### ISSUE TITLE
Connection between learning rate and total number of training epochs

### ISSUE THREAD

**Issue by eypros** [ultralytics/yolov5#11475]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I was training a Yolov5n model on Coco dataset using the default parameters with the exception of _batch size 64_ (I could not afford higher in my machine). Anyway, after the training ended I got a mAP50 of _45.09%_ which I found acceptable though it was obviously less than _45.7%_ (aka the reported precision). I figured out that the smaller batch size might impact the performance (as mentioned in the issues).

Then, I tried to train for more epochs without changing anything to the model's parameters. At first the mAP50 for both experiments coincided in large degree (to my satisfaction) but after some point there was a small at first and then large diversion between the precisions.

first epochs:
![image](https://user-images.githubusercontent.com/31985256/235857853-227f12e1-ff3a-48dd-8d30-2fd807abdd21.png)

next epochs:
![image](https://user-images.githubusercontent.com/31985256/235859571-01d3aa85-9e76-4475-8709-40fe824bcadb.png)
In the previous plots the blue is the initial run and with the black line is the second attempt.

My question, seeing the aforementioned behavior, is whether the actual learning rate is affected by the total number of epochs. As I investigated the code I think I didn't found any connection, meaning that the learning rate would be the same up to epoch 300 (the final epoch of the first run). If this is the case, is it normal to observe such diversions in mAP between different runs?

**Comment by glenn-jocher** [comment]
@eypros hi!

Great question! From analyzing the graphs, it seems like the model's performance has plateaued around epochs 250-300, and continuing to train the model past that point hasn't been helpful in improving performance.

As for your question about learning rate, the learning rate schedule actually decays the learning rate over time as the model gets closer to convergence. Here's an example of a learning rate schedule from the `train.py` script:

```python
optimizer = torch.optim.SGD(model.parameters(), lr=hyp['lr0'], momentum=hyp['momentum'], nesterov=True)
lf = lambda x: ((1 + math.cos(x * math.pi / hyp['epochs'])) / 2) * (1 - hyp['lrf']) + hyp['lrf']  # cosine
scheduler = lr_scheduler.LambdaLR(optimizer, lr_lambda=lf)
```

The `lr_lambda` function used by the PyTorch learning rate scheduler decays the learning rate as the model gets closer to convergence. In this case, the learning rate takes on a cosine function whose minimum is `hyp['lrf']` and whose maximum is `hyp['lr0']`.

To answer your question, if `train.py` has been used to train the model and the `lf` schedule is used, then the learning rate will already be decaying over time, which means that the number of epochs shouldn't cause sudden dives in mAP.

Hope that helps!
